include/tss2/tss2_esys.h

```diff
@@ -3302,6 +3404,17 @@ Esys_GetSysContext(
     ESYS_CONTEXT *esys_context,
     TSS2_SYS_CONTEXT **sys_context);
 
+TSS2_RC
+Esys_SetCryptoCallbacks(
+    ESYS_CONTEXT *esysContext,
+    ESYS_CRYPTO_CALLBACKS *callbacks,
+    void *userdata);
+
+ESYS_TR
+Esys_GetCryptoCallbacks(
```

Comment:
What is the purpose of GetCallbacks ?
IMHO a setter is enough, no need for the getter.

Reply:
I was initially leaning towards that, but then I was thinking about signal handlers that give you the current state so you can back up and restore. So in this case you could figure out at runtime if t he backend has no crypto linked and set your own, or back up and restore something for some reason.

Reply:
I guess this somewhat makes sense.
BTW, was happens, if the user calls this without setting before if the build-in thing is set ?
Are the esys-pointers returned or is null returned ?

Reply:
> I guess this somewhat makes sense. BTW, was happens, if the user calls this without setting before if the build-in thing is set ? Are the esys-pointers returned or is null returned ?

Depends:
| backend | return |
| ----------- | -------- |
| ossl        | ossl esys pointers |
| mbed     | mbed esys pointers |
| none      | null |

So they can always restore what was originally configured.

Reply:
I do see the use case of when you want to override only one of the crypto functions but use the default ones for all the rest, I guess.
But this could also be achieved by leaving the pointers inside the struct as NULL. (btw, is this supported)

In order to restore the original pointers, I though you could just call the setter with a NULL pointer instead of a struct ?

Reply:
ok with me